zone/spell_effects.cpp

```diff
@@ -1820,6 +1820,63 @@ bool Mob::SpellEffect(Mob* caster, uint16 spell_id, float partial, int level_ove
 
 				break;
 			}
+			case SE_SummonCorpseZone:
+			{
+#ifdef SPELL_EFFECT_SPAM
+				snprintf(effect_desc, _EDLEN, "Summon Corpse Zone");
+#endif
+				if (!IsNPC()) {
+					Client* client_target = this->CastToClient();
+					Group* group = entity_list.GetGroupByClient(client_target);
+					if(group) {
+						if(!group->IsGroupMember(client_target)) {
+							if (caster != this) {
+								caster->Message(Chat::Red, "Your target must be a group member for this spell.");
+								break;
+							}
+						}
+					} else if (caster) {
+						Raid *raid = entity_list.GetRaidByClient(caster->CastToClient());
+						if(raid) {
+							uint32 group_id = 0xFFFFFFFF;
+							group_id = raid->GetGroup(caster->GetName());
```

Comment:
Can we inline `group_id` here

```uint32 group_id = raid->GetGroup(caster->GetName()); ```

`GetGroup` returns `0xFFFFFFFF` if there is no group anyways, I'd prefer if we did a check for `group_id` > 0 and `group_id` < MAX_RAID_GROUPS because we can get overflowed values back (4294967295), if we ever had this return 0 or -1 the current code would not work correctly